Source/santad/EventProviders/EndpointSecurity/Enricher.mm

```diff
@@ -86,6 +87,36 @@
       return std::make_unique<EnrichedMessage>(
         EnrichedCSInvalidated(std::move(es_msg), Enrich(*es_msg->process)));
 #if HAVE_MACOS_13
+    case ES_EVENT_TYPE_NOTIFY_AUTHENTICATION:
+      switch (es_msg->event.authentication->type) {
+        case ES_AUTHENTICATION_TYPE_OD:
+          return std::make_unique<EnrichedMessage>(
+            EnrichedAuthenticationOD(std::move(es_msg), Enrich(*es_msg->process),
+                                     Enrich(es_msg->event.authentication->data.od->instigator)));
+        case ES_AUTHENTICATION_TYPE_TOUCHID:
+          return std::make_unique<EnrichedMessage>(EnrichedAuthenticationTouchID(
+            std::move(es_msg), Enrich(*es_msg->process),
+            Enrich(es_msg->event.authentication->data.touchid->instigator),
+            es_msg->event.authentication->data.touchid->has_uid
+              ? UsernameForUID(es_msg->event.authentication->data.touchid->uid.uid)
+              : std::nullopt));
+        case ES_AUTHENTICATION_TYPE_TOKEN:
+          return std::make_unique<EnrichedMessage>(EnrichedAuthenticationToken(
+            std::move(es_msg), Enrich(*es_msg->process),
+            Enrich(es_msg->event.authentication->data.token->instigator)));
+        case ES_AUTHENTICATION_TYPE_AUTO_UNLOCK:
+          return std::make_unique<EnrichedMessage>(EnrichedAuthenticationAutoUnlock(
+            std::move(es_msg), Enrich(*es_msg->process),
+            UIDForUsername(
+              StringTokenToStringView(es_msg->event.authentication->data.auto_unlock->username))));
+
+        // Note: return nullptr here rather than exit. Unlike the programmer error case
+        // below which results in an exit, here we have a case of potentially new
+        // authentication types being added in the future for this one event type. If
+        // that happens, we don't want versions of Santa compiled for an older SDK to
```

Comment:
nit: This makes it sounds like building with a new SDK automatically adds support for new auth sub types. In practice Santa needs code changes, in addition to building with a new SDK.

Reply:
Updated comment, PTAL